packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2835,7 +2889,29 @@ class TextSelectionGestureDetector extends StatefulWidget {
 }
 
 class _TextSelectionGestureDetectorState extends State<TextSelectionGestureDetector> {
-  static int? _getDefaultMaxConsecutiveTap() => 2;
+  static int? _getDefaultMaxConsecutiveTap() {
+    switch(defaultTargetPlatform) {
+      // From observation, these platforms reset their tap count to 0 when the number of consecutive taps
+      // exceeds 3. For example on Linux, when going past a triple click, on the fourth click the selection
+      // is moved to the precise click position, on the fifth click the word at the position is selected, and
+      // on the sixth click the paragraph at the position is selected.
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.linux:
+        return 3;
+      // From observation, these platforms either hold their tap count at 3 or they
+      // allow it to grow infinitely. For example on macOS, when going past a triple click,
+      // the selection should be retained at the paragraph that was first selected on triple click.
+      case TargetPlatform.iOS:
+      case TargetPlatform.macOS:
+      // From observation, this platforms consecutive tap count grows infinitely, alternating
+      // between double click and triple click actions. For example, after a triple click has
+      // selected a paragraph, on the next click the word at the clicked position will be selected,
+      // and on the next click the paragraph at the position.
+      case TargetPlatform.windows:
```

Comment:
Why do iOS and Mac fall through to the same behavior as Windows if the comments above seem to describe their behavior differently?

Reply:
To me it looked like iOS/macOS/Windows increased their tap count infinitely but they handled taps past 3 differently.

iOS and macOS don't handle anything past a third tap. The selection is retained at the paragraph that was selected on the triple tap.

Windows alternates between double click and triple click when going past the third tap. Alternating between selecting the word or paragraph at the position.

Let me know if I can clarify this further.

Reply:
I think it's ok as-is.  I looked through the code to see how consecutiveTapCount is used and it makes sense that these all return null but later have different behavior.

Reply:
I think it makes more sense now after introducing an `effectiveTapCount`.